src/main/java/frc/robot/subsystems/drivetrain/DriveBase.java

```diff
@@ -1,24 +1,48 @@
 package frc.robot.subsystems.drivetrain;
 
 import edu.wpi.first.math.controller.SimpleMotorFeedforward;
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.kinematics.DifferentialDriveOdometry;
+import edu.wpi.first.math.util.Units;
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
 
 public class DriveBase extends SubsystemBase {
     // Static final for now, make sure to change in order to switch between drives
-    private static final double wheelRadiusMeters = 0.0508;
-    private static final double maxVelocityMetersPerSec = 10;
+    private static final double wheelRadiusMeters = Units.inchesToMeters(2);
+    private static final double maxVelocityMetersPerSec = Units.feetToMeters(16.5);
+    private static final double trackWidthMeters = 0.61;
 
     private final DriveIO driveIO;
 
     private final SimpleMotorFeedforward leftFF;
     private final SimpleMotorFeedforward rightFF;
 
+    private DifferentialDriveOdometry odometry =
+        new DifferentialDriveOdometry(new Rotation2d(),
+            new Pose2d());
+
     public DriveBase(DriveIO driveIO) {
         this.driveIO = driveIO;
 
         // Arbitrary values for now
-        leftFF = new SimpleMotorFeedforward(0.01, 0.01);
-        rightFF = new SimpleMotorFeedforward(0.01, 0.01);
+        this.leftFF = new SimpleMotorFeedforward(
+            driveIO.getkS(),
+            driveIO.getkV(),
+            driveIO.getkA());
+        this.rightFF = new SimpleMotorFeedforward(
+            driveIO.getkS(),
+            driveIO.getkV(),
+            driveIO.getkA());
+        
+        setBrakeMode(false);
+    }
+
+    @Override
+    public void periodic() {
+        odometry.update(new Rotation2d(driveIO.getGyroAngle()),
```

Comment:
I believe that the `getGyroAngle()` returns the opposite sign
```suggestion
        odometry.update(new Rotation2d(-driveIO.getGyroAngle()),
```